packages/react/src/hooks/useW3IAccount.ts

```diff
@@ -75,30 +64,33 @@ export const useW3iAccount = (address?: string): W3iAccountReturn => {
 
       throw new Error("Web3InboxClient not ready");
     },
-    [web3inboxClientData, account]
+    [clientData, account]
   );
 
   const unregister = useCallback(async () => {
-    if (web3inboxClientData && account) {
-      return web3inboxClientData.client.unregister({ account });
+    if (clientData && account) {
+      return clientData.client.unregister({ account });
     }
-  }, [web3inboxClientData, account]);
+  }, [clientData, account]);
 
   useEffect(() => {
-    if (!address) return;
+    if (!address || !clientData?.client) return;
+    setAccount(`eip155:1:${address}`);
```

Comment:
@devceline I think we'll need to move this logic to the outside of the `useW3IAccount` hook since they might want to set accounts for different networks. Not just `eip155:1:{address}`.

In this case, we should give better feedback to the user in case they forget to set account on connect. So we could remove this `useEffect` from here.

Reply:
@devceline Can you review this? How/where should be account set logic do you think?